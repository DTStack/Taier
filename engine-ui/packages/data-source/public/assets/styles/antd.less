@import '~antd/dist/antd.less';

@primary-color: #3f87ff; // 全局主色
@success-color: #12bc6a; // 成功色
@warning-color: #ffb310; // 警告色
@error-color: #ff5f5c; // 错误色
@font-size-base: 12px; // 主字号
@link-color: #5c99ff; // 链接色
@heading-color: #333; // 标题色
@text-color: #333; // 主文本色
@text-color-secondary: #666; // 次文本色
@disabled-color: #bfbfbf; // 失效色
@border-color-base: #ddd; // 边框色
// @box-shadow-base: 0 2px 8px rgba(0, 0, 0, 0.15); // 浮层阴影

// antd 分页器样式
.ant-pagination-item-active {
	a {
		font-family: PingFangSC-Regular;
		letter-spacing: 0;
		text-align: center;
	}
}

.ant-pagination {
	position: relative;
	left: 9px;
	box-sizing: border-box;
	margin: 0;
	margin: 14px 10px 0 0;
	padding: 0;
	color: rgba(0, 0, 0, 0.85);
	font-size: 14px;
	font-variant: tabular-nums;
	line-height: 1.5715;
	list-style: none;
	font-feature-settings: 'tnum';
}

.ant-table-cell {
	border-right: none !important;
}
.ant-table-measure-row td > div {
	display: none !important;
}
.dt-notification-close-btn {
	position: absolute !important;
	top: 18px !important;
	right: 46px;
	font-weight: normal;
	font-size: 13px;
	line-height: 17px;
	user-select: none;
	&:hover {
		color: rgb(39, 39, 39);
	}
}
.anticon-eye-invisible .ant-input-password-icon {
	color: #999;
}
.anticon-close .ant-select-remove-icon {
	color: #999;
}

// 输入框，下拉框禁用样式
.ant-input[disabled] {
	color: #999999 !important;
	background-color: #e8e8e8 !important;
	border: 1px solid #dddddd;
}

.ant-select-disabled {
	&:hover {
		border: none;
	}
	.ant-select-selection {
		color: #999999 !important;
		background-color: #e8e8e8 !important;
		border: 1px solid #dddddd;
	}
}

.ant-form-item {
	margin-bottom: 20px;

	&.ant-form-item-with-help {
		margin-bottom: 4px;
	}

	label {
		top: 50%;
		display: inline-block;
		transform: translateY(-50%);
	}

	.ant-form-item-label {
		height: 32px;
	}

	.ant-form-item-label,
	.ant-form-item-control {
		line-height: unset;
	}

	textarea.ant-input {
		margin-bottom: 0;
	}
}

.ant-modal {
	height: 80vh;

	.ant-modal-content {
		max-height: 80vh;
	}

	.ant-modal-body {
		max-height: calc(80vh - 127px);
		padding-bottom: 28px;
		overflow-y: auto;
	}

	.ant-modal-footer,
	.ant-modal-body,
	.ant-modal-header {
		padding-right: 20px;
		padding-left: 20px;
		border-color: #ddd;
	}

	.ant-modal-footer {
		padding: 19.5px 16px;
	}
}

.ant-table-tbody > tr > td {
	vertical-align: middle;
}

.ant-btn:hover,
.ant-btn:focus,
.ant-btn:active,
.ant-btn.active {
	color: #333;
	border-color: #bfbfbf;
}
.ant-btn.ant-btn-primary:hover,
.ant-btn.ant-btn-primary:focus {
	color: #fff;
	background-color: #5c99ff;
	border-color: #5c99ff;
}
.ant-btn.ant-btn-background-ghost:hover,
.ant-btn.ant-btn-background-ghost:focus {
	color: #fff;
	background-color: #5c99ff !important;
	border-color: #5c99ff;
}

.ant-message-notice-content {
	height: auto !important;
	.ant-message-custom-content {
		height: auto !important;
	}
}
