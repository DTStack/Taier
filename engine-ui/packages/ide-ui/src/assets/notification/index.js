"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _rcNotification = _interopRequireDefault(require("rc-notification"));

var _icon = _interopRequireDefault(require("antd/lib/icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var notificationInstance = {};
var defaultDuration = 4.5;
var defaultTop = 24;
var defaultBottom = 24;
var defaultPlacement = 'topRight';
var defaultGetContainer;
var defaultCloseIcon;

function setNotificationConfig(options) {
  var duration = options.duration,
      placement = options.placement,
      bottom = options.bottom,
      top = options.top,
      getContainer = options.getContainer || (() => document.querySelector('div[id="app"]')),
      closeIcon = options.closeIcon;

  if (duration !== undefined) {
    defaultDuration = duration;
  }

  if (placement !== undefined) {
    defaultPlacement = placement;
  }

  if (bottom !== undefined) {
    defaultBottom = bottom;
  }

  if (top !== undefined) {
    defaultTop = top;
  }

  if (getContainer !== undefined) {
    defaultGetContainer = getContainer;
  }

  if (closeIcon !== undefined) {
    defaultCloseIcon = closeIcon;
  }
}

function getPlacementStyle(placement) {
  var top = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultTop;
  var bottom = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultBottom;
  var style;

  switch (placement) {
    case 'topLeft':
      style = {
        left: 0,
        top: top,
        bottom: 'auto'
      };
      break;

    case 'topRight':
      style = {
        right: 0,
        top: top,
        bottom: 'auto'
      };
      break;

    case 'bottomLeft':
      style = {
        left: 0,
        top: 'auto',
        bottom: bottom
      };
      break;

    default:
      style = {
        right: 0,
        top: 'auto',
        bottom: bottom
      };
      break;
  }

  return style;
}

function getNotificationInstance(_ref, callback) {
  var prefixCls = _ref.prefixCls,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? defaultPlacement : _ref$placement,
      _ref$getContainer = _ref.getContainer,
      getContainer = _ref$getContainer === void 0 ? defaultGetContainer : _ref$getContainer,
      top = _ref.top,
      bottom = _ref.bottom,
      _ref$closeIcon = _ref.closeIcon,
      closeIcon = _ref$closeIcon === void 0 ? defaultCloseIcon : _ref$closeIcon;
  var cacheKey = "".concat(prefixCls, "-").concat(placement);

  if (notificationInstance[cacheKey]) {
    callback(notificationInstance[cacheKey]);
    return;
  }

  var closeIconToRender = React.createElement("span", {
    className: "".concat(prefixCls, "-close-x")
  }, closeIcon || React.createElement(_icon["default"], {
    className: "".concat(prefixCls, "-close-icon"),
    type: "close"
  }));

  _rcNotification["default"].newInstance({
    prefixCls: prefixCls,
    className: "".concat(prefixCls, "-").concat(placement),
    style: getPlacementStyle(placement, top, bottom),
    getContainer: getContainer,
    closeIcon: closeIconToRender
  }, function (notification) {
    notificationInstance[cacheKey] = notification;
    callback(notification);
  });
}

var typeToIcon = {
  success: 'check-circle-o',
  info: 'info-circle-o',
  error: 'close-circle-o',
  warning: 'exclamation-circle-o'
};

function notice(args) {
  var outerPrefixCls = args.prefixCls || 'ide-ui-notification';
  var prefixCls = "".concat(outerPrefixCls, "-notice");
  var duration = args.duration === undefined ? defaultDuration : args.duration;
  var iconNode = null;

  if (args.icon) {
    iconNode = React.createElement("span", {
      className: "".concat(prefixCls, "-icon")
    }, args.icon);
  } else if (args.type) {
    var iconType = typeToIcon[args.type];
    iconNode = React.createElement(_icon["default"], {
      className: "".concat(prefixCls, "-icon ").concat(prefixCls, "-icon-").concat(args.type),
      type: iconType
    });
  }

  var autoMarginTag = !args.description && iconNode ? React.createElement("span", {
    className: "".concat(prefixCls, "-message-single-line-auto-margin")
  }) : null;
  var placement = args.placement,
      top = args.top,
      bottom = args.bottom,
      getContainer = args.getContainer,
      closeIcon = args.closeIcon;
  getNotificationInstance({
    prefixCls: outerPrefixCls,
    placement: placement,
    top: top,
    bottom: bottom,
    getContainer: getContainer,
    closeIcon: closeIcon
  }, function (notification) {
    notification.notice({
      content: React.createElement("div", {
        className: iconNode ? "".concat(prefixCls, "-with-icon") : ''
      }, iconNode, React.createElement("div", {
        className: "".concat(prefixCls, "-message")
      }, autoMarginTag, args.message), React.createElement("div", {
        className: "".concat(prefixCls, "-description")
      }, args.description), args.btn ? React.createElement("span", {
        className: "".concat(prefixCls, "-btn")
      }, args.btn) : null),
      duration: duration,
      closable: true,
      onClose: args.onClose,
      onClick: args.onClick,
      key: args.key,
      style: args.style || {},
      className: args.className
    });
  });
}

var api = {
  open: notice,
  close: function close(key) {
    Object.keys(notificationInstance).forEach(function (cacheKey) {
      return notificationInstance[cacheKey].removeNotice(key);
    });
  },
  config: setNotificationConfig,
  destroy: function destroy() {
    Object.keys(notificationInstance).forEach(function (cacheKey) {
      notificationInstance[cacheKey].destroy();
      delete notificationInstance[cacheKey];
    });
  }
};
['success', 'info', 'warning', 'error'].forEach(function (type) {
  api[type] = function (args) {
    return api.open(_extends(_extends({}, args), {
      type: type
    }));
  };
});
api.warn = api.warning;
var _default = api;
exports["default"] = _default;
//# sourceMappingURL=index.js.map
