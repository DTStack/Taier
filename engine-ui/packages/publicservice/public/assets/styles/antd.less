@import '~antd/dist/antd.less';
@main-color: #268BFC;

// antd 分页器样式
.ant-pagination-item {
  &:hover {
    border-color: @main-color  !important;

    a {
      color: @main-color  !important;
    }
  }
}

.ant-pagination-item-active {
  background: @main-color  !important;

  a {
    font-family: PingFangSC-Regular;
    font-size: 14px;
    color: #ffffff !important;
    letter-spacing: 0;
    text-align: center;
  }

  &:hover {
    border-color: @main-color  !important;

    a {
      color: #ffffff !important;
    }
  }
}

.ant-pagination {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  position: relative;
  left: 9px;
  color: rgba(0, 0, 0, 0.85);
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  list-style: none;
  font-feature-settings: 'tnum';
  // margin: 10px;
  margin: 14px 10px 0 0;
}

.ant-table-cell {

  border-right: none !important;

}
.ant-table-measure-row td>div{
	display: none !important;
}