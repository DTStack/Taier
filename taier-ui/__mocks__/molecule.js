const { editor, ...rest } = require('./monaco-editor');

module.exports = {
    Component: class {
        emit = jest.fn();
        subscribe = jest.fn();
        getState = () => this.state;
        setState = (state) => Object.assign(this.state, state);
    },
    FolderTreeService: class FolderTreeService {
        get = jest.fn();
    },
    GlobalEvent: class {
        emit = jest.fn();
        subscribe = jest.fn();
    },
    panel: {
        getState: jest.fn(),
        setState: jest.fn(),
        setActive: jest.fn(),
        add: jest.fn(),
        update: jest.fn(),
        getPanel: jest.fn(),
    },
    explorer: {
        forceUpdate: jest.fn(),
    },
    editor: {
        updateActions: jest.fn(),
        updateGroup: jest.fn(),
        updateTab: jest.fn(),
        getDefaultActions: jest.fn(),
        getState: jest.fn(),
        getGroupIdByTab: jest.fn(),
        setActive: jest.fn(),
        isOpened: jest.fn(),
        open: jest.fn(),
        closeTab: jest.fn(),
        ...editor,
    },
    folderTree: {
        get: jest.fn(),
        add: jest.fn(),
        getState: jest.fn(),
        update: jest.fn(),
        setActive: jest.fn(),
    },
    layout: {
        getState: jest.fn(),
        togglePanelVisibility: jest.fn(),
    },
    Header: ({ title, toolbar }) => (
        <div data-testid="mockHeader">
            {title}
            <div data-testid="mockHeaderToolbar">{toolbar}</div>
        </div>
    ),
    Content: ({ children }) => <div data-testid="mockContent">{children}</div>,
    FolderTree: jest.fn(),
    connect: jest.fn((_, children) => children),
    TreeViewUtil: jest.fn(),
    FileTypes: {},
    TreeNodeModel: class {
        constructor(params) {
            Object.assign(this, params);
        }
    },
    colorTheme: {
        getColorThemeMode: jest.fn(),
        onChange: jest.fn(),
    },
    notification: {
        getState: jest.fn(),
        add: jest.fn(),
    },
    component: {
        Scrollbar: ({ children }) => <div data-testid="Scrollbar">{children}</div>,
        Icon: ({ type }) => <svg data-testid="Icon" type={type} />,
        ActionBar: ({ data = [] }) => (
            <div data-testid="ActionBar">
                {data.map((i) => (
                    <div key={i.id} id={i.id}>
                        {i.name}
                    </div>
                ))}
            </div>
        ),
    },
    sidebar: {
        setActive: jest.fn(),
    },
    activityBar: {
        setActive: jest.fn(),
    },
    ...rest,
};
