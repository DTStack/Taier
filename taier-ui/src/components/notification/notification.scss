@import "@/styles/variables.scss";

.dt-notification.ant-notification-notice {
    padding: 5px 13px;
    cursor: pointer;

    .ant-notification-notice-message {
        margin-bottom: 0;
        overflow: hidden;
        font-size: $font-size;
        white-space: nowrap;
        text-overflow: ellipsis;
    }

    .ant-notification-notice-close {
        top: 6px;
        right: 14px;
    }

    &:hover {
        background-color: var(--button-hoverBackground);
    }
}
