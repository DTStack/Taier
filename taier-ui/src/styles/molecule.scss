@import "./variables.scss";

.mo-menuBar--horizontal {
    height: $menuBar-size;
    background: var(--activityBar-background);

    > .mo-menu {
        height: $menuBar-size;
        color: var(--activityBar-inactiveForeground);
        background: inherit;
    }
}

// only the tabs in editor should have hover styles
.mo-editor__group {
    .mo-tab__item:hover {
        background-color: var(--tab-hoverBackground);
    }
}

.mo-tab__item__status,
.mo-tab__item__op {
    color: inherit;

    &:hover {
        color: inherit;
    }
}

.mo-editor__group-breadcrumb {
    color: var(--breadcrumb-foreground);

    .mo-breadcrumb__item:hover {
        color: var(--breadcrumb-focusForeground);
        opacity: 1;
    }
}

.mo-notification__item:focus {
    background: var(--list-hoverBackground);
    outline: 1px solid var(--list-focusOutline);
}

.mo-statusBar__item > a:empty {
    padding: 0;
}

// ===================== Auxiliary Bar ===========================
.mo-auxiliaryBar__tabs {
    .mo-auxiliaryBar__tabs__tab {
        font-size: 12px;
        padding-top: 5px;
        padding-bottom: 5px;

        &:hover {
            color: var(--taier-rightBar-foreground);
            background-color: var(--taier-rightBar-background);
        }

        &--active {
            color: var(--taier-rightBar-foreground);
            background-color: var(--taier-rightBar-background);
            position: relative;

            &::after {
                position: absolute;
                top: 0;
                right: 0;
                bottom: 0;
                width: 2px;
                height: 100%;
                background-color: #3f87ff;
                content: "";
            }
        }
    }
}

.mo-auxiliaryBar__container {
    background: var(--sideBar-background);
}
