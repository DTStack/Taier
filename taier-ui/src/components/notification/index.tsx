import { CheckCircleOutlined, CloseCircleOutlined } from '@ant-design/icons';
import molecule from '@dtinsight/molecule';
import { notification as antNotification } from 'antd';
import type { ArgsProps } from 'antd/lib/notification';
import './notification.scss';

type INotificationConfigs = Pick<ArgsProps, 'message'> & { key: string };

class Notification {
    private assertNotExistNotification(key: string) {
        const { data } = molecule.notification.getState();
        return !data?.find((n) => n.id === key);
    }
    private showNotification() {
        const { showNotifications } = molecule.notification.getState();
        if (!showNotifications) {
            molecule.notification.toggleNotification();
        }
    }
    private highlightNotification(key: string) {
        const notificationItem = document.getElementById(key)?.parentElement;
        if (notificationItem) {
            notificationItem.tabIndex = 0;
            notificationItem.focus();
        }
    }
    /**
     * Add a notification both in antd and molecule
     */
    openWithMolecule(config: INotificationConfigs) {
        const { showNotifications } = molecule.notification.getState();
        // the antd's notification pops up only when the molecule's notification invisible
        if (!showNotifications) {
            antNotification.open({
                key: config.key,
                message: config.message,
                placement: 'bottomRight',
                className: 'dt-notification',
                duration: 8,
                closeIcon: <></>,
                onClose() {},
                onClick: () => {
                    antNotification.close(config.key);
                    this.showNotification();
                    setTimeout(() => {
                        this.highlightNotification(config.key);
                    }, 0);
                },
            });
        }

        if (this.assertNotExistNotification(config.key)) {
            molecule.notification.add([
                {
                    id: config.key,
                    value: '',
                    render() {
                        return config.message;
                    },
                },
            ]);
        }
    }

    /**
     * Open a notification both in antd and molecule in bottomRight with danger icon
     */
    error({ key, message }: { key: string; message: string }) {
        this.openWithMolecule({
            key,
            message: (
                <>
                    <CloseCircleOutlined
                        style={{
                            color: 'var(--editorError-foreground)',
                            marginRight: 5,
                        }}
                    />
                    <span id={key} title={message}>
                        {message}
                    </span>
                </>
            ),
        });
    }

    success({ key, message }: { key: string; message: string }) {
        this.openWithMolecule({
            key,
            message: (
                <>
                    <CheckCircleOutlined
                        style={{
                            color: 'var(--terminal-ansiBrightGreen)',
                            marginRight: 5,
                        }}
                    />
                    <span id={key} title={message}>
                        {message}
                    </span>
                </>
            ),
        });
    }
}
export default new Notification();
