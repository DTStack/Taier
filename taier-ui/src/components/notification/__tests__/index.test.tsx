import molecule from '@dtinsight/molecule';
import notification from '..';
import { notification as antNotification } from 'antd';
import { cleanup, render } from '@testing-library/react';

jest.mock('antd', () => ({
    notification: {
        open: jest.fn(),
    },
}));

describe('Test Notification Component', () => {
    beforeEach(() => {
        cleanup();
        document.body.innerHTML = '';
    });
    it("Should pop up an antd's notification", () => {
        (molecule.notification.getState as jest.Mock).mockReset().mockImplementation(() => ({
            showNotifications: false,
        }));

        notification.openWithMolecule({ message: 'test', key: '1' });

        expect(antNotification.open).toBeCalledTimes(1);
    });

    it('Should not render duplicated notification', () => {
        (molecule.notification.getState as jest.Mock)
            .mockReset()
            .mockImplementationOnce(() => ({
                showNotifications: true,
                data: [],
            }))
            .mockImplementation(() => ({
                showNotifications: true,
                data: [{ id: '1' }],
            }));

        notification.openWithMolecule({ message: 'test', key: '1' });
        expect(molecule.notification.add).toBeCalledTimes(1);

        notification.openWithMolecule({ message: 'test', key: '1' });
        // Still called one time since duplicated key notification should be prevented
        expect(molecule.notification.add).toBeCalledTimes(1);
    });

    it('Should match snapshots', () => {
        (molecule.notification.getState as jest.Mock).mockReset().mockImplementation(() => ({
            showNotifications: true,
            data: [],
        }));

        (molecule.notification.add as jest.Mock).mockReset();

        notification.error({ message: 'test', key: '1' });

        const message = (molecule.notification.add as jest.Mock).mock.calls[0][0][0].render();
        expect(render(<>{message}</>).asFragment()).toMatchSnapshot();

        notification.success({ message: 'test', key: '1' });
        expect(
            render(<>{(molecule.notification.add as jest.Mock).mock.calls[1][0][0].render()}</>).asFragment()
        ).toMatchSnapshot();
    });
});
