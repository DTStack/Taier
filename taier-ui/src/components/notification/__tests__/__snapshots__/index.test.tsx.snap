// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Notification Component Should match snapshots 1`] = `
<DocumentFragment>
  <span
    aria-label="close-circle"
    class="anticon anticon-close-circle"
    role="img"
    style="margin-right: 5px;"
  >
    <svg
      aria-hidden="true"
      data-icon="close-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
      />
      <path
        d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
      />
    </svg>
  </span>
  <span
    id="1"
    title="test"
  >
    test
  </span>
</DocumentFragment>
`;

exports[`Test Notification Component Should match snapshots 2`] = `
<DocumentFragment>
  <span
    aria-label="check-circle"
    class="anticon anticon-check-circle"
    role="img"
    style="margin-right: 5px;"
  >
    <svg
      aria-hidden="true"
      data-icon="check-circle"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0051.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z"
      />
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
      />
    </svg>
  </span>
  <span
    id="1"
    title="test"
  >
    test
  </span>
</DocumentFragment>
`;
